src/readonly-table/index.ts

```diff
@@ -1,220 +1,221 @@
+          <thead class="kuc-readonly-table__table__header">
+            <tr>
+              ${this.columns.map((column) => this._getColumnsTemplate(column))}
+            </tr>
+          </thead>
+          <tbody class="kuc-readonly-table__table__body">
+            ${currentPageData.map((data: object, currentIndex: number) => {
+              return this._getDataTemplate(data, currentIndex);
+            })}
+          </tbody>
+        </table>
+        <kuc-base-pagination
+          .visible="${this.pagination}"
+          .isPrev="${this._toggleDisplayPreviousButton()}"
+          .isNext="${this._toggleDisplayNextButton()}"
+          @kuc:pagination-click-prev=${this._handleClickPreviousButton}
+          @kuc:pagination-click-next=${this._handleClickNextButton}
+        ></kuc-base-pagination>
+      `;
+    }
 
-  update(changedProperties: PropertyValues) {
-    if (changedProperties.has("columns")) this._validateColumns(this.columns);
-    if (changedProperties.has("data")) this._validateData(this.data);
+    private _createDisplayData(
+      data: object[],
+      pagePosition: number,
+      rowsPerPage: number
+    ) {
+      if (!this.pagination) return data;
+      const firstRow = (pagePosition - 1) * rowsPerPage + 1;
+      const lastRow = pagePosition * rowsPerPage;
+      const displayData = data.filter(
+        (_element, index: number) =>
+          index >= firstRow - 1 && index <= lastRow - 1
+      );
+      return displayData;
+    }
 
-    super.update(changedProperties);
-  }
+    private _getColumnsTemplate(column: Column) {
+      return html`
+        <th
+          class="kuc-readonly-table__table__header__cell"
+          ?hidden="${column.visible === false}"
+        >
+          <span class="kuc-readonly-table__table__header__cell__label"
+            >${column.title}</span
+          >
+        </th>
+      `;
+    }
 
-  private _getColumnsTemplate(column: Column) {
-    return html`
-      <th
-        class="kuc-readonly-table__table__header__cell"
-        ?hidden="${column.visible === false}"
-      >
-        <span class="kuc-readonly-table__table__header__cell__label">
-          ${column.header && column.header.text}</span
+    // Formatting the data displayed on the current page
+    private _getDataTemplate(data: any, currentIndex: number) {
+      return html`
+        <tr
+          class="kuc-readonly-table__table__body__row kuc-readonly-table__table__body__row-${currentIndex}"
         >
-      </th>
-    `;
-  }
+          ${this._columnOrder.map((currentCol, colIndex) => {
+            let isHidden = false;
+            if (
+              this.columns[colIndex] &&
+              this.columns[colIndex].visible === false
+            ) {
+              isHidden = true;
+            }
+            const value = data[currentCol];
+            /* eslint-disable */
+            return html`
+              <td class="kuc-readonly-table__table__body__row__cell-data">${value}</td>
```

Comment:
It seem you missing ?hidden="${!visible}" in <td> tag to hidden column.
please check it